bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerOpenOp.java

```diff
@@ -39,21 +43,21 @@
  * Encapsulates the ledger open operation
  *
  */
-class LedgerOpenOp implements GenericCallback<LedgerMetadata> {
+class LedgerOpenOp implements GenericCallback<LedgerMetadata>, OpenBuilder {
     static final Logger LOG = LoggerFactory.getLogger(LedgerOpenOp.class);
 
     final BookKeeper bk;
-    final long ledgerId;
```

Comment:
separate the builder code from the op code. It serves a different end and makes the business logic less clear.

Reply:
@ivankelly  done